tests/queries/0_stateless/02241_remote_filesystem_cache_on_insert.reference

```diff
@@ -41,25 +41,25 @@ file_segment_range_end:   1659
 size:                     1660
 state:                    DOWNLOADED
 SELECT count() FROM (SELECT arrayJoin(cache_paths) AS cache_path, local_path, remote_path FROM system.remote_data_paths ) AS data_paths INNER JOIN system.filesystem_cache AS caches ON data_paths.cache_path = caches.cache_path;
-7
+1
 SELECT count() FROM system.filesystem_cache;
-7
+1
 SELECT count() FROM system.filesystem_cache;
-7
+1
 INSERT INTO test SELECT number, toString(number) FROM numbers(100) SETTINGS enable_filesystem_cache_on_write_operations=0;
 SELECT count() FROM system.filesystem_cache;
-7
+1
 INSERT INTO test SELECT number, toString(number) FROM numbers(100);
 INSERT INTO test SELECT number, toString(number) FROM numbers(300, 10000);
 SELECT count() FROM system.filesystem_cache;
-21
+3
 OPTIMIZE TABLE test FINAL;
 SELECT count() FROM system.filesystem_cache;
-27
+6
 SET mutations_sync=2;
 ALTER TABLE test UPDATE value = 'kek' WHERE key = 100;
 SELECT count() FROM system.filesystem_cache;
-28
+6
```

Comment:
These values must actually differ (line 58 and 62), second one should be bigger as in previous `.reference`. I haven't looked at your code yet, but may be a setting `enable_filesystem_cache_on_wrote_operations` in settings or `cache_on_write_operations` in config was turned on for you locally, as it can be the reason for these values not being updated.

Reply:
> These values must actually differ (line 58 and 62), second one should be bigger as in previous `.reference`. I haven't looked at your code yet, but may be a setting `enable_filesystem_cache_on_wrote_operations` in settings or `cache_on_write_operations` in config was turned off for you locally, as it can be the reason for these values not being updated.

I did turn on the cache-on-write configuration, because the following is the output after I write, you can see that the data already exists in the local cache after writing

`INSERT INTO test SELECT number, toString(number) FROM numbers(100);
SELECT file_segment_range_begin, file_segment_range_end, size, state FROM (SELECT file_segment_range_begin, file_segment_range_end, size, state, local_path FROM (SELECT arrayJoin(cache_paths) AS cache_path, local_path, remote_path FROM system.remote_data_paths ) AS data_paths INNER JOIN system.filesystem_cache AS caches ON data_paths.cache_path = caches.cache_path) WHERE endsWith(local_path, 'data.bin') FORMAT Vertical;
Row 1:
──────
file_segment_range_begin: 0
file_segment_range_end:   745
size:                     746
state:                    DOWNLOADED`

Reply:
> These values must actually differ (line 58 and 62), second one should be bigger as in previous `.reference`. I haven't looked at your code yet, but may be a setting `enable_filesystem_cache_on_wrote_operations` in settings or `cache_on_write_operations` in config was turned off for you locally, as it can be the reason for these values not being updated.

The reason is that I turned on the **cache_enabled** option, which caused the metadata cache to be saved by DiskCacheWrapper. After closing it, the problem was solved, thank you very much.